src/server/json_family.cc

```diff
@@ -269,6 +269,25 @@ facade::OpStatus SetJson(const OpArgs& op_args, string_view key, JsonType&& valu
   return OpStatus::OK;
 }
 
+size_t GetJsonArrayIndex(int index, size_t size) {
+  if (index >= 0) {
+    return index;
+  }
+
+  if (static_cast<size_t>(-index) > size) {
+    return 0;
+  }
```

Comment:
Just making sure, we're supposed to fall back to the first element if index is out of range? Is that what Redis does?

Reply:
Also what if `size` is `0`? That will result in an invalid index, no?

Reply:
Yes, that's exactly what Redis does. For example, if the array has a length of 5 and the user sends -55, the normalized index will be 0.

Also, all added tests are a request and a response that I copy from Redis.